meteor/server/api/rundownLayouts.ts

```diff
@@ -61,22 +65,10 @@ PickerPOST.route('/shelfLayouts/upload/:showStyleBaseId', async (params, req: In
 
 	let content = ''
 	try {
-		if (!showStyleBase) throw new Meteor.Error(404, `ShowStylebase "${showStyleBaseId}" not found`)
-
-		const body = req.body
-		if (!body) throw new Meteor.Error(400, 'Restore Shelf Layout: Missing request body')
-
-		if (typeof body !== 'string' || body.length < 10)
-			throw new Meteor.Error(400, 'Restore Shelf Layout: Invalid request body')
+		if (!showStyleBase) throw new Meteor.Error(404, `Show Style Base "${showStyleBaseId}" not found`)
 
-		const layout = JSON.parse(body) as RundownLayoutBase
-		check(layout._id, Match.Optional(String))
-		check(layout.name, String)
-		check(layout.type, String)
-
-		layout.showStyleBaseId = showStyleBase._id
-
-		await RundownLayouts.upsertAsync(layout._id, layout)
+		const body = validateAndGetRestoredLayoutBody(req)
```

Comment:
Since this is refined to the type `RundownLayoutBase`, `body` could be renamed to something more descriptive.